docs/proposals/mqtt.md

```diff
@@ -0,0 +1,234 @@
+# MQTT Protocol Implementation
+
+## Goal
+Agent implements [MQTT Protocol](https://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.html), a light-weight
+publish/subscribe messaging protocol widely used in Internet of Things (IoT).
+
+Supporting MQTT as a protocol in Akri provides a mechanism by which Application Developers (IoT, Kubernetes)
+can provide scalable applications based on current status of the IoT Devices within an environment. 
+
+Depending on the IoT Device's availability the application pertaining to the data such a device produces
+can be scaled accordingly.
+
+## What can Akri solve?
```

Comment:
Should this be narrowed based on the discovery strategy. For example, Akri + MQTT Broker discovery makes it so an operator can say, run this client (Akri "broker") on nodes that can see an MQTT Broker with x,y, and z topics. How do MQTT clients normally respond to topics no longer being published (devices going away)? It seems like with this implementation, if a topic goes away that a client/akri broker was using, Akri would automatically bring down the client. Is that ideal?

Reply:
> How do MQTT clients normally respond to topics no longer being published (devices going away)?
There is a possibility of maintaining the connection session between the akri broker (containing mqtt client) however to my knowledge there is no mechanism in MQTT that provides a graceful disconnection if there is no data being published on the subscribed topics. I have to the check the MQTTv5 Specifications where they provide [Session Expiry and Message Expiry](https://www.hivemq.com/blog/mqtt5-essentials-part4-session-and-message-expiry/).
The Session Expiry feature might be able to trigger a graceful disconnection, which in turn leads to bringing down the Akri broker in a much more dynamic manner.